jmh-benchmarks/src/main/java/org/apache/kafka/jmh/record/RecordBatchIterationBenchmark.java

```diff
@@ -142,15 +169,15 @@ private ByteBuffer createBatch(int batchSize) {
                     break;
             }
 
-            builder.append(0, null, value);
+            builder.append(0, null, value, headers);
         }
         return builder.build().buffer();
     }
 
     @Benchmark
     public void measureValidation(Blackhole bh) throws IOException {
         MemoryRecords records = MemoryRecords.readableRecords(singleBatchBuffer.duplicate());
-        LogValidator.validateMessagesAndAssignOffsetsCompressed(records, new TopicPartition("a", 0),
+        LogValidator.validateMessagesAndAssignOffsets(records, new TopicPartition("a", 0),
```

Comment:
Why did we change this?

Reply:
> Why did we change this?

run JMH with uncompressed data.

The alternative way is that we can add an new benchmark for ```assignOffsetsNonCompressed```. However, the test data of this class includes both uncompressed and compressed so it is weird to input compressed data to test ```assignOffsetsNonCompressed```.

Reply:
Right, but the goal of this benchmark was to test the compressed case where there is additional overhead due to compression buffer allocation and so on. As you say, it would be better to have another benchmark for the uncompressed case.

Reply:
> it would be better to have another benchmark for the uncompressed case.

I creates two new classes - ```CompressedRecordBatchValidationBenchmark``` and ```UncompressedRecordBatchValidationBenchmark```. They use different compression type and measure different validation function.